src/adam_bashforth_adaptive.jl

```diff
@@ -0,0 +1,339 @@
+    dy = typeof(F(t[1], y[1]))[]
+    push!(y,F(t[1], y[1]))
+
+    #initialization of step size array, which will have variable length as well
+    h = initstep
+    next_h = 0.0
+    if h == 0.
+        # initial guess at a step size
+        h, tdir, dy0 = hinit(F, y0, tstart, tfinal, 3, reltol, abstol)
+    else
+        tdir = sign(tfinal - tstart)
+    end
+
+    #uncomment line below to test with fixed step size
+    #@show h = diff(tspan)
+
+    #initialization of arrays used in calculation of further steps
+    #all initialized to a order x order array, which will need to expand to
+    # n x order over course of algorithm, as we use n steps, where n is
+    #unknown
+    Tdy=typeof(F(t[1], y[1]))
+    c,g,b,Φstar, Φ = intialize_coef_arrays(max_order+3,Tdy)
+
+    #Array which holds norms of LE(n+1), the local error or truncation error, which will
+    #will used in step size selection. They correspond to norm of LE(n+1)_{k-1},
+    # LE(n+1)_{k}, E(n+1)_{k+1}, and LE(n+1)_{k+2} from Harier pgs 422-424
+    err_km1, err_k, err_kp1, err_kp2 = 0.0, 0.0, 0.0, 0.0
+
+    ##Note that due to Julia starting indexing at 1, k = steporder + 1 in what follows
+    n = 1     #`n` counts the time steps. We start at n=1, the first step
+    steporder=1 #`steporder` is the order of the AB method at the current step
+
+    timeout_const = 0 # reset for how many steps before a new step size can be taken
+    timeout = timeout_const # counter for this over the following loop
+
+    wait=0 #`wait` is used to control the order adaptivity, because the order
+           # can be increased at most every other step.
+
+    consec_const_steps = 0 #counts how many consecutives steps taken with same step size
+
+    while  tfinal*tdir > t[n]*tdir #&&  minstep < abs(h)
+        # Need to run the first several steps at reduced order, and need to maintain
+        # n>steporder invariant in order to perform adaptive step selection of the
+        # 'order' = steporder, for it requires the last 'order' number of points
+        if n<=order+1
+            steporder = n==1 ? 1 : min(n-1, order)
+        end
+
+        #allocate space for t[n+1], y[n+1], dy[n+1], next_h
+        increment_length!(t,y,dy,n)
+
+        #Current derivative: set dy[n] = F(t[n], y[n])
+        dy[n]=F(t[n], y[n])
+
+        successful_step = false
+        while !successful_step
+          t[n+1]=t[n]+h
+
+          #####################################################################
+          #Calculating explicit approximatiion to y[n+1]
+          #####################################################################
+          #calculation of c_{j,q} coefficients for j=0,...,k-1; q = 0,...,k-1-(j-1)
+          g_coefs!(g,c,h,t,n,min(n,steporder+2))
+
+          #calculation of b_j(n), Φstar_j(n), and Φ_j(n) coefficients from recurence relations
+          Φ_and_Φstar_coefs!(b, Φ, Φstar,dy[n],t,n,1:min(max(n-1,1),steporder+1))
```

Comment:
maybe assign `1:min(max(n-1,1),steporder+1)` to a variable, so its clear what it is.